### PR TITLE
Time-based rewards

### PR DESCRIPTION
**Pull request by daemon**
Fixes aforementioned segfault.

### CODE REVIEW

File: src/hscore/hscore_rewards.c [hunk 1]
```diff
@@ -471,21 +471,21 @@ local void flagWinCallback(Arena *arena, int freq, int *pts)
 			if (flagging_freq(arena, i->p_freq))
```
Comment (Ceiu) [inline src/hscore/hscore_rewards.c:471]: Are the ship & human checks missing from this line?

File: src/hscore/hscore_rewards.c [hunk 2]
```diff
@@ -471,21 +471,21 @@ local void flagWinCallback(Arena *arena, int freq, int *pts)
 			if (flagging_freq(arena, i->p_freq))
 				end_playtime(i, i->p_freq);
 
-		ticks_t q60_playtime = 1; // Mathematically 60th quartile, realistically 80th for small games (< 5)
+		ticks_t q60_playtime = 1; // Mathematically 60th percentile, realistically 80th for small games (< 5)
 		size_t n_flagging = 0;
 		FOR_EACH_PLAYER_IN_ARENA(i, arena)
-			if (flagging_freq(arena, i->p_freq) && i->p_freq == freq)
-				++n_flagging;	
+			if (flagging_freq(arena, i->p_freq) && i->p_freq == freq && IS_HUMAN(i))
```
Comment (Ceiu) [inline src/hscore/hscore_rewards.c]: Should we be checking the ship here as well so we don't pick up spec/chatnet clients?

File: src/hscore/hscore_rewards.c [hunk 3]
```diff
@@ -471,21 +471,21 @@ local void flagWinCallback(Arena *arena, int freq, int *pts)
...
+				++n_flagging;
 
 		if (n_flagging > 0)
 		{
 			ticks_t *flagging_times = amalloc(sizeof(ticks_t) * n_flagging);
 			int j = 0;
 			FOR_EACH_PLAYER_IN_ARENA(i, arena)
-				if (flagging_freq(arena, i->p_freq) && i->p_freq == freq)
+				if (flagging_freq(arena, i->p_freq) && i->p_freq == freq && IS_HUMAN(i))
```
Comment (Ceiu) [inline src/hscore/hscore_rewards.c]: Same here (ship check)

File: src/hscore/hscore_rewards.c [hunk 4]
```diff
@@ -569,7 +572,7 @@ local void flagWinCallback(Arena *arena, int freq, int *pts)
 
 		pd->Lock();
 		FOR_EACH_PLAYER_IN_ARENA(i, arena)
-			if (flagging_freq(arena, i->p_freq))
+			if (flagging_freq(arena, i->p_freq) && IS_HUMAN(i))
```
Comment (Ceiu) [inline src/hscore/hscore_rewards.c:575]: Ship check?

File: src/hscore/hscore_rewards.c [hunk 5]
```diff
@@ -879,11 +882,11 @@ local void shipFreqChangeCallback(Player *p, int newship, int oldship, int newfr
 	PData *pdata = PPDATA(p, pdkey);
 	pdata->periodic_tally = 0;
 
-	if (flagging_freq(p->arena, oldfreq) && !flagging_freq(p->arena, newfreq))
+	if (flagging_freq(p->arena, oldfreq) && !flagging_freq(p->arena, newfreq) && IS_HUMAN(p))
```
Comment (Ceiu) [inline src/hscore/hscore_rewards.c:885]: Ship checks?

**Comment by daemon** [comment]
Okay, it'll be a few hours before I'll be able to test these additional ship check changes.

**Comment by Ceiu** [comment]
Looks good; tests on stage went well.

**Pull request merged**